### PR TITLE
SQLAlchemy 2.0

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR updates SQLAlchemy to 2.0.

### CODE REVIEW

File: app/handlers/base.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ def Session(self):
             # must merge the user object with the current session
             # ref: https://docs.sqlalchemy.org/en/14/orm/session_basics.html#adding-new-or-existing-items
             session.add(self.current_user)
+            session.bind = DBSession.session_factory.kw["bind"]
```
Comment (stefanv) [inline app/handlers/base.py:145]: Looks good, thanks Michael. What is this line for?

Comment (mcoughlin) [reply app/handlers/base.py:145]: @stefanv I find that the explicit bind is necessary in this new 2.0 setup!

Comment (guynir42) [reply app/handlers/base.py:145]: what is actually the value of this `kw["bind"]`?

**Review by guynir42 — COMMENTED** [review]
This is a big step that we should not take lightly. 
Did you try to run this with all of skyportal's tests?

**Comment by mcoughlin** [comment]
@guynir42 It is Engine(postgresql://skyportal:***@localhost:5432/skyportal)

**Comment by Theodlz** [comment]
This looks good. I played around with the associated SkyPortal PR and I didn't notice any misbehavior so far.
At least for the Baselayer part of this SQLAlchemy update, it deserves a LGTM!

**Comment by mcoughlin** [comment]
Ok so @Theodlz is happy and we can most easily test effects of sqlalchemy in skyportal. Will pin this there.

**Comment by stefanv** [comment]
Could you please also patch up the template_app so it runs without warnings under 2.0

**Pull request merged**